sql: TestTelemetry failed

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2174280&tab=buildLog) on [master@44c362600dbed9d58be8daf64558543745698736](https://github.com/cockroachdb/cockroach/commits/44c362600dbed9d58be8daf64558543745698736):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (4.95s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry371069144
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (2.16s)
        datadriven.go:194: 
            testdata/telemetry/planning:160: SELECT * FROM (VALUES (1), (2)) AS a(x) JOIN LATERAL (SELECT a.x+1 AS x) AS b ON a.x < b.x
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2174505&tab=buildLog) on [master@ed56390d4f28f7d7b26ce047bcb799afb337b537](https://github.com/cockroachdb/cockroach/commits/ed56390d4f28f7d7b26ce047bcb799afb337b537):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (42.07s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry795625072
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (20.72s)
        datadriven.go:194: 
            testdata/telemetry/planning:160: SELECT * FROM (VALUES (1), (2)) AS a(x) JOIN LATERAL (SELECT a.x+1 AS x) AS b ON a.x < b.x
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2174851&tab=buildLog) on [master@239805ade7d2e2e2628266fefc818ac60dcf74dc](https://github.com/cockroachdb/cockroach/commits/239805ade7d2e2e2628266fefc818ac60dcf74dc):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (28.52s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry350729575
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (13.89s)
        datadriven.go:194: 
            testdata/telemetry/planning:170: SELECT x FROM (VALUES (1)) AS b(x) WHERE EXISTS(SELECT * FROM (VALUES (1)) AS a(x) WHERE a.x = b.x)
            expected:
            sql.plan.subquery.correlated
            
            found:
            sql.plan.cte
            sql.plan.subquery.correlated

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2174988&tab=buildLog) on [master@be8c1a17b940ecc56dc3521ab8264f662c88072c](https://github.com/cockroachdb/cockroach/commits/be8c1a17b940ecc56dc3521ab8264f662c88072c):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (28.16s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry990182758
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (12.99s)
        datadriven.go:194: 
            testdata/telemetry/planning:160: SELECT * FROM (VALUES (1), (2)) AS a(x) JOIN LATERAL (SELECT a.x+1 AS x) AS b ON a.x < b.x
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2175074&tab=buildLog) on [master@9803ab8458d2b8b1b1abc63f32a49d216b126137](https://github.com/cockroachdb/cockroach/commits/9803ab8458d2b8b1b1abc63f32a49d216b126137):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (28.00s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry136006042
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (12.95s)
        datadriven.go:194: 
            testdata/telemetry/planning:155: SELECT * FROM (VALUES (1), (2)) AS a(x), LATERAL (SELECT a.x+1)
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2175218&tab=buildLog) on [master@89208ec8b5350fd34036b559a258403e820deda6](https://github.com/cockroachdb/cockroach/commits/89208ec8b5350fd34036b559a258403e820deda6):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (28.38s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry932612632
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (12.89s)
        datadriven.go:194: 
            testdata/telemetry/planning:155: SELECT * FROM (VALUES (1), (2)) AS a(x), LATERAL (SELECT a.x+1)
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2175492&tab=buildLog) on [master@52b8eb1b1c90d1ff9c8faae8c80d45b7de56d350](https://github.com/cockroachdb/cockroach/commits/52b8eb1b1c90d1ff9c8faae8c80d45b7de56d350):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (27.95s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry540540927
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (12.98s)
        datadriven.go:194: 
            testdata/telemetry/planning:160: SELECT * FROM (VALUES (1), (2)) AS a(x) JOIN LATERAL (SELECT a.x+1 AS x) AS b ON a.x < b.x
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2175690&tab=buildLog) on [master@28c0337df4b695b6eba09639ca312a92348b9294](https://github.com/cockroachdb/cockroach/commits/28c0337df4b695b6eba09639ca312a92348b9294):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (28.51s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry036539023
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (13.21s)
        datadriven.go:194: 
            testdata/telemetry/planning:155: SELECT * FROM (VALUES (1), (2)) AS a(x), LATERAL (SELECT a.x+1)
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2176185&tab=buildLog) on [master@fc9b2d13bae4449bfbeb8ad0aa496a0892793440](https://github.com/cockroachdb/cockroach/commits/fc9b2d13bae4449bfbeb8ad0aa496a0892793440):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (5.84s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry804671736
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (2.16s)
        datadriven.go:194: 
            testdata/telemetry/planning:155: SELECT * FROM (VALUES (1), (2)) AS a(x), LATERAL (SELECT a.x+1)
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2175800&tab=buildLog) on [master@a165626f1a0f74480db0741112b736defe5a3e87](https://github.com/cockroachdb/cockroach/commits/a165626f1a0f74480db0741112b736defe5a3e87):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (30.18s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry494580315
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (14.07s)
        datadriven.go:194: 
            testdata/telemetry/planning:160: SELECT * FROM (VALUES (1), (2)) AS a(x) JOIN LATERAL (SELECT a.x+1 AS x) AS b ON a.x < b.x
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2175877&tab=buildLog) on [master@7831db43748b1df410510c954283d8fb4e0d4694](https://github.com/cockroachdb/cockroach/commits/7831db43748b1df410510c954283d8fb4e0d4694):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (42.24s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry597503449
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (20.25s)
        datadriven.go:194: 
            testdata/telemetry/planning:170: SELECT x FROM (VALUES (1)) AS b(x) WHERE EXISTS(SELECT * FROM (VALUES (1)) AS a(x) WHERE a.x = b.x)
            expected:
            sql.plan.subquery.correlated
            
            found:
            sql.plan.cte
            sql.plan.subquery.correlated

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql).TestTelemetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2175941&tab=buildLog) on [master@bbbfdbd1919c6de411742c8442cfc3903d33ee86](https://github.com/cockroachdb/cockroach/commits/bbbfdbd1919c6de411742c8442cfc3903d33ee86):

```
=== RUN   TestTelemetry
--- FAIL: TestTelemetry (30.28s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestTelemetry514175508
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestTelemetry/planning
    --- FAIL: TestTelemetry/planning (14.05s)
        datadriven.go:194: 
            testdata/telemetry/planning:155: SELECT * FROM (VALUES (1), (2)) AS a(x), LATERAL (SELECT a.x+1)
            expected:
            sql.plan.lateral-join
            
            found:
            sql.plan.cte
            sql.plan.lateral-join

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestTelemetry PKG=./pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestTelemetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/testdata/telemetry/planning
+++ b/pkg/sql/testdata/telemetry/planning
@@ -137,8 +137,6 @@ sql.plan.stats.created
 # Test various planning counters.
 feature-allowlist
 sql.plan.cte.*
-sql.plan.lateral-join
-sql.plan.subquery.*
 ----
 
 feature-usage
@@ -151,6 +149,11 @@ WITH RECURSIVE a AS (SELECT 1 UNION ALL SELECT * FROM a WHERE false) SELECT * FR
 ----
 sql.plan.cte
 sql.plan.cte.recursive
+
+feature-allowlist
+sql.plan.lateral-join
+sql.plan.subquery.*
+----
 
 feature-usage
 SELECT * FROM (VALUES (1), (2)) AS a(x), LATERAL (SELECT a.x+1)

